charts/promtail/templates/service-extra.yaml

```diff
@@ -7,8 +7,8 @@ metadata:
   namespace: {{ include "promtail.namespaceName" $ }}
   labels:
     {{- include "promtail.labels" $ | nindent 4 }}
-    {{- with .labels }}
-    {{- toYaml $ | nindent 4 }}
+    {{- with $values.labels }}
```

Comment:
Have you tested it ?

Reply:
Hi @Sheikh-Abubaker yes I did, here an extract from an environment where I have deployed this:
```
promtail:
  extraPorts:
    Fab1:
      name: fab1
      labels:
        type: bgp-ingress
      containerPort: 1513
      protocol: TCP
      service:
        type: LoadBalancer
        port: 1514
```

```
kgs aci-mon-stack-promtail-fab1 --show-labels
NAME                          TYPE           CLUSTER-IP      EXTERNAL-IP      PORT(S)          AGE     LABELS
aci-mon-stack-promtail-fab1   LoadBalancer   10.109.206.89   192.168.12.135   1514:30646/TCP   3d17h   app.kubernetes.io/instance=aci-mon-stack,app.kubernetes.io/managed-by=Helm,app.kubernetes.io/name=promtail,app.kubernetes.io/version=3.0.0,helm.sh/chart=promtail-6.16.3-camrossi,type=bgp-ingress
```

See the last label.

Cheers!

Reply:
LGTM!